### PR TITLE
Fix recurring tasks

### PR DESCRIPTION
**Pull request by Nadrieril**

### CODE REVIEW

**Comment by Nadrieril** [comment]
I have now rebased this on top of https://github.com/Ekleog/todiff/pull/17, so https://github.com/Ekleog/todiff/pull/17 need to be merged first

File: tests/tests.yaml [hunk 1]
```diff
@@ -270,21 +291,19 @@ simple_strict_recurrence:
...
   allowed_divergence: 50
   from:
-    - 2018-06-01 foo due:2018-06-20 rec:1m
+    - 2018-06-01 foo due:2018-06-20 rec:+1m
 
   to:
-    - x 2018-06-17 2018-06-01 foo due:2018-06-15 rec:1m
-    - 2018-06-17 foo due:2018-07-15 rec:1m
+    - x 2018-06-17 2018-06-01 foo due:2018-06-15 rec:+1m
+    - 2018-06-17 foo due:2018-07-15 rec:+1m
```
Comment (Ekleog) [inline tests/tests.yaml]: Hmm… I think with this change we're missing a test case for non-strict recurrence.

For instance,
```
2018-06-01 foo due:2018-06-20 rec:1m
```
to
```
x 2018-07-04 2018-06-01 foo due:2018-06-20 rec:1m
2018-07-04 foo due:2018-08-04 rec:1m
```
(as just confirmed using `topydo` to manually complete the task)

File: src/compute_changes.rs [hunk 2]
```diff
@@ -370,28 +377,33 @@ pub fn compute_changeset(
 
     let changes = matches
         .into_iter()
-        .map(|x| {
-            let new_delta = match &x.delta {
+        .map(|ChangedTask{ orig, delta }| {
```
Comment (Ekleog) [inline src/compute_changes.rs]: Oh, and this is missing a space between `ChangedTask` and `{`, according to `cargo fmt`.

File: tests/tests.yaml (hunk 1 above)
Comment (Nadrieril) [reply tests/tests.yaml]: Thanks for the example !

**Comment by Nadrieril** [comment]
Done

**Comment by Ekleog** [comment]
Great, thanks! merged :+1:

**Pull request merged**